Not compiling with libdns v1.0.0 due to changes to Record interface

Errors like:

> 149.1 # github.com/libdns/mythicbeasts
149.1 /go/pkg/mod/github.com/libdns/mythicbeasts@v1.0.1/client.go:97:64: record.Type undefined (type libdns.Record has no field or method Type)
149.1 /go/pkg/mod/github.com/libdns/mythicbeasts@v1.0.1/client.go:97:83: record.Name undefined (type libdns.Record has no field or method Name)
149.1 /go/pkg/mod/github.com/libdns/mythicbeasts@v1.0.1/client.go:97:103: record.Value undefined (type libdns.Record has no field or method Value)
149.1 /go/pkg/mod/github.com/libdns/mythicbeasts@v1.0.1/client.go:97:126: record.TTL undefined (type libdns.Record has no field or method TTL)
149.1 /go/pkg/mod/github.com/libdns/mythicbeasts@v1.0.1/client.go:160:64: record.Type undefined (type libdns.Record has no field or method Type)
149.1 /go/pkg/mod/github.com/libdns/mythicbeasts@v1.0.1/client.go:160:83: record.Name undefined (type libdns.Record has no field or method Name)
149.1 /go/pkg/mod/github.com/libdns/mythicbeasts@v1.0.1/client.go:160:103: record.Value undefined (type libdns.Record has no field or method Value)
149.1 /go/pkg/mod/github.com/libdns/mythicbeasts@v1.0.1/client.go:160:126: record.TTL undefined (type libdns.Record has no field or method TTL)
149.1 /go/pkg/mod/github.com/libdns/mythicbeasts@v1.0.1/client.go:174:10: record.Name undefined (type libdns.Record has no field or method Name)
149.1 /go/pkg/mod/github.com/libdns/mythicbeasts@v1.0.1/client.go:175:10: record.Type undefined (type libdns.Record has no field or method Type)
149.1 /go/pkg/mod/github.com/libdns/mythicbeasts@v1.0.1/client.go:175:10: too many errors

According to [libdns/libdns/releases/tag/v1.0.0](https://github.com/libdns/libdns/releases/tag/v1.0.0)
>[Record](https://pkg.go.dev/github.com/libdns/libdns#Record) is now an interface, not a struct type.
[RR](https://pkg.go.dev/github.com/libdns/libdns#RR) is a new struct type that is similar to what Record was, but is a generic representation of a "Resource Record".

This occured during an xcaddy build of caddy. Also see [caddyserver/caddy/releases/tag/v2.10.0](https://github.com/caddyserver/caddy/releases/tag/v2.10.0):
>libdns 1.0 APIs: Many of you use [DNS provider modules](https://github.com/caddy-dns) to solve ACME DNS challenges or to enable dynamic DNS. They implement interfaces defined by [libdns](https://github.com/libdns/libdns) to get, set, append, and delete DNS records. After 5 years of production experience, including lessons learned with ECH, libdns APIs have been updated and 1.0 beta has been tagged. DNS provider packages will need to update their code to be compatible, which will help ensure stability and well-defined semantics for the future. Several packages have already updated or are in the process of updating (cloudflare, rfc2136, and desec to name a few).

I don't have much go experience but I will see if I can make these changes as they appear to be straightforward.

--- a/_example/main.go
+++ b/_example/main.go
@@ -3,6 +3,7 @@ package main
 import (
 	"context"
 	"fmt"
+	"net/netip"
 	"time"
 
 	"github.com/libdns/libdns"
@@ -24,8 +25,8 @@ func main() {
 
 	// Append Records Test
 	recordsAdded, err := provider.AppendRecords(ctx, zone, []libdns.Record{
-		{Type: "A", Name: "test1", Value: "8.8.4.4", TTL: time.Duration(123) * time.Second},
-		{Type: "CNAME", Name: "test2", Value: "www.example.com.", TTL: time.Duration(666) * time.Second},
+		libdns.Address{Name: "test1", IP: netip.MustParseAddr("8.8.4.4"), TTL: time.Duration(123) * time.Second},
+		libdns.CNAME{Name: "test2", Target: "www.example.com.", TTL: time.Duration(666) * time.Second},
 	})
 	if err != nil {
 		fmt.Printf("ERROR: %s\n", err.Error())
@@ -33,9 +34,9 @@ func main() {
 
 	// Set Records Test
 	recordsSet, err := provider.SetRecords(ctx, zone, []libdns.Record{
-		{Type: "A", Name: "test1", Value: "8.8.8.8", TTL: time.Duration(999) * time.Second},
-		{Type: "CNAME", Name: "test2", Value: "test2.example.com", TTL: time.Duration(999) * time.Second},
-		{Type: "CNAME", Name: "test3", Value: "test3.example.net"},
+		libdns.Address{Name: "test1", IP: netip.MustParseAddr("8.8.8.8"), TTL: time.Duration(999) * time.Second},
+		libdns.CNAME{Name: "test2", Target: "test2.example.com", TTL: time.Duration(999) * time.Second},
+		libdns.CNAME{Name: "test3", Target: "test3.example.net"},
 	})
 	if err != nil {
 		fmt.Printf("ERROR: %s\n", err.Error())
@@ -43,8 +44,8 @@ func main() {
 
 	// Delete Records Test
 	recordsDeleted, err := provider.DeleteRecords(ctx, zone, []libdns.Record{
-		{Type: "A", Name: "test1"},
-		{Type: "CNAME", Name: "test2"},
+		libdns.Address{Name: "test1"},
+		libdns.CNAME{Name: "test2"},
 	})
 	if err != nil {
 		fmt.Printf("ERROR: %s\n", err.Error())

--- a/client.go
+++ b/client.go
@@ -93,8 +93,9 @@ func (p *Provider) addRecord(ctx context.Context, zone string, record libdns.Rec
 
 	var addedRecords []libdns.Record
 
+	rr := record.RR()
 	data := mythicRecords{}
-	data.Records = append(data.Records, mythicRecord{Type: record.Type, Name: record.Name, Value: record.Value, TTL: int(record.TTL.Seconds())})
+	data.Records = append(data.Records, mythicRecord{Type: rr.Type, Name: rr.Name, Value: rr.Data, TTL: int(rr.TTL.Seconds())})
 
 	payload, err := json.Marshal(data)
 	if err != nil {
@@ -156,8 +157,9 @@ func (p *Provider) updateRecord(ctx context.Context, zone string, record libdns.
 
 	var updatedRecords []libdns.Record
 
+	rr := record.RR()
 	data := mythicRecords{}
-	data.Records = append(data.Records, mythicRecord{Type: record.Type, Name: record.Name, Value: record.Value, TTL: int(record.TTL.Seconds())})
+	data.Records = append(data.Records, mythicRecord{Type: rr.Type, Name: rr.Name, Value: rr.Data, TTL: int(rr.TTL.Seconds())})
 
 	payload, err := json.Marshal(data)
 
@@ -171,8 +173,8 @@ func (p *Provider) updateRecord(ctx context.Context, zone string, record libdns.
 		"/zones/"+
 		zone+
 		"/records/"+
-		record.Name+"/"+
-		record.Type+
+		rr.Name+"/"+
+		rr.Type+
 		"?exclude-template&exclude-generated", body)
 	if err != nil {
 		return nil, fmt.Errorf("addRecord: Error in NewRequestWithContext: %s", err.Error())
@@ -228,8 +230,9 @@ func (p *Provider) removeRecord(ctx context.Context, zone string, record libdns.
 
 	var removedRecords []libdns.Record
 
+	rr := record.RR()
 	data := mythicRecords{}
-	data.Records = append(data.Records, mythicRecord{Type: record.Type, Name: record.Name, Value: record.Value, TTL: int(record.TTL.Seconds())})
+	data.Records = append(data.Records, mythicRecord{Type: rr.Type, Name: rr.Name, Value: rr.Data, TTL: int(rr.TTL.Seconds())})
 
 	payload, err := json.Marshal(data)
 	if err != nil {
@@ -241,8 +244,8 @@ func (p *Provider) removeRecord(ctx context.Context, zone string, record libdns.
 		"/zones/"+
 		zone+
 		"/records/"+
-		record.Name+"/"+
-		record.Type+
+		rr.Name+"/"+
+		rr.Type+
 		"?exclude-template&exclude-generated", body)
 	if err != nil {
 		return nil, fmt.Errorf("addRecord: Error in NewRequestWithContext: %s", err.Error())

--- a/go.mod
+++ b/go.mod
@@ -3,6 +3,6 @@ module github.com/libdns/mythicbeasts
 go 1.16
 
 require (
-	github.com/libdns/libdns v0.2.1
+	github.com/libdns/libdns v1.0.0
 	golang.org/x/net v0.0.0-20220531201128-c960675eff93
 )

--- a/go.mod
+++ b/go.mod
@@ -3,6 +3,6 @@ module github.com/libdns/mythicbeasts
 go 1.16
 
 require (
-	github.com/libdns/libdns v1.0.0
+	github.com/libdns/libdns v1.0.0-beta.1
 	golang.org/x/net v0.0.0-20220531201128-c960675eff93
 )

--- a/provider.go
+++ b/provider.go
@@ -68,12 +68,17 @@ func (p *Provider) GetRecords(ctx context.Context, zone string) ([]libdns.Record
 	var records []libdns.Record
 
 	for _, r := range result.Records {
-		records = append(records, libdns.RR{
+		record, err := libdns.RR{
 			Type: r.Type,
 			Name: r.Name,
 			Data: r.Value,
 			TTL:  time.Duration(r.TTL) * time.Second,
-		})
+		}.Parse()
+		if err != nil {
+			return nil, fmt.Errorf("GetRecords: failed to parse record %s: %d", r.Name, err)
+		}
+
+		records = append(records, record)
 	}
 	return records, nil
 }

--- a/provider.go
+++ b/provider.go
@@ -68,11 +68,11 @@ func (p *Provider) GetRecords(ctx context.Context, zone string) ([]libdns.Record
 	var records []libdns.Record
 
 	for _, r := range result.Records {
-		records = append(records, libdns.Record{
-			Type:  r.Type,
-			Name:  r.Name,
-			Value: r.Value,
-			TTL:   time.Duration(r.TTL) * time.Second,
+		records = append(records, libdns.RR{
+			Type: r.Type,
+			Name: r.Name,
+			Data: r.Value,
+			TTL:  time.Duration(r.TTL) * time.Second,
 		})
 	}
 	return records, nil

